nx/lib/nx.ex

```diff
@@ -5244,6 +5244,38 @@ defmodule Nx do
     impl!(tensor).conjugate(%{tensor | type: Nx.Type.to_complex(tensor.type)}, tensor)
   end
 
+  @doc """
+  Calculates the complex phase angle of each element in the tensor.
+
+  If $z = a + bi = r e^\\theta$, $phase(z) = {\rm atan2}(b, a)$
+
+  ## Examples
+
+       iex> Nx.phase(Complex.new(1, 2))
+       #Nx.Tensor<
+         c64
+         1.1071487665176392+0.0i
+       >
+
+       iex> Nx.phase(1)
+       #Nx.Tensor<
+         f32
+         0.0
+       >
+
+       iex> Nx.phase(Nx.tensor([Complex.new(1, 2), Complex.new(-2, 1)]))
```

Comment:
We can also start using `~V` and `~M`, since main depends on Elixir v1.13.

```suggestion
       iex> Nx.phase(~V[1+2i -2+1i])
```

WDYT @polvalente?